### PR TITLE
feat: override support for api category

### PR DESCRIPTION
**Pull request by ammarkarachi**
**Feature: Use Amplify maintained, easy to use types to override Amplify-generated AppSync Apis for ** 
How to use?
1. “amplify  init”  to create a new Amplify project
2. “amplify  add api” to add graphql api 
3. “amplify  override api”,to override  the  AppSync API stack
4. “amplify  push”  to deploy!

### CODE REVIEW

File: packages/amplify-category-api/resources/awscloudformation/overrides-resource/APIGW/package.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  "name": "overrides",
+  "version": "1.0.0",
+  "description": "",
+  "scripts": {
+    "build": "tsc",
+    "watch": "tsc -w",
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "dependencies": {
+    "@aws-amplify/cli-extensibility-helper": "2.3.0-apiext3.0"
```
Comment (edwardfoyle) [inline packages/amplify-category-api/resources/awscloudformation/overrides-resource/APIGW/package.json:11]: should this be a non-tagged version?

File: packages/amplify-category-api/resources/awscloudformation/overrides-resource/AppSync/package.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "overrides",
+  "version": "1.0.0",
+  "description": "",
+  "scripts": {
+    "build": "tsc",
+    "watch": "tsc -w"
+  },
+  "dependencies": {
+    "@aws-amplify/cli-extensibility-helper": "2.3.0-apiext3.0"
```
Comment (edwardfoyle) [inline packages/amplify-category-api/resources/awscloudformation/overrides-resource/AppSync/package.json:10]: same here

Comment (kaustavghosh06) [reply packages/amplify-category-api/resources/awscloudformation/overrides-resource/AppSync/package.json:10]: Yes, should be "^2.0.0" like other categories (auth and storage).

File: packages/amplify-category-api/resources/awscloudformation/overrides-resource/APIGW/package.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "overrides",
+  "version": "1.0.0",
+  "description": "",
+  "scripts": {
+    "build": "tsc",
+    "watch": "tsc -w",
+    "test": "echo \"Error: no test specified\" && exit 1"
```
Comment (edwardfoyle) [inline packages/amplify-category-api/resources/awscloudformation/overrides-resource/APIGW/package.json:8]: remove this

File: packages/amplify-category-api/resources/awscloudformation/overrides-resource/AppSync/package.json (hunk 2 above)
Comment (jhockett) [reply packages/amplify-category-api/resources/awscloudformation/overrides-resource/AppSync/package.json:10]: We haven't published a version with API types included. A tagged version is necessary on the first release.

File: packages/amplify-category-api/resources/awscloudformation/overrides-resource/APIGW/package.json (hunk 3 above)
Comment (jhockett) [reply packages/amplify-category-api/resources/awscloudformation/overrides-resource/APIGW/package.json:8]:
```suggestion
```

Comment (jhockett) [inline packages/amplify-category-api/resources/awscloudformation/overrides-resource/APIGW/package.json:8]:
```suggestion
    "watch": "tsc -w"
```

File: packages/amplify-category-api/resources/awscloudformation/overrides-resource/AppSync/package.json (hunk 2 above)
Comment (kaustavghosh06) [reply packages/amplify-category-api/resources/awscloudformation/overrides-resource/AppSync/package.json:10]: But, when we publish to prod/release ,"@aws-amplify/cli-extensibility-helper" will be published will API types.

File: packages/amplify-category-api/src/index.ts [hunk 4]
```diff
@@ -180,55 +198,72 @@ export async function getPermissionPolicies(context, resourceOpsMapping) {
...
   } else {
     commandPath = path.join(commandPath, category, context.input.command);
   }
 
-  const commandModule = require(commandPath);
-  await commandModule.run(context);
+  const commandModule = await import(commandPath);
+  try {
+    await commandModule.run(context);
+  } catch (error) {
```
Comment (edwardfoyle) [inline packages/amplify-category-api/src/index.ts:223]: aren't errors caught and handled higher up the call stack? Doesn't seem like we need a try/catch here

File: packages/amplify-category-api/src/provider-utils/awscloudformation/api-input-manager/appsync-api-input-state.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
+import { AmplifyCategories, AmplifySupportedService, CLIInputSchemaValidator, JSONUtilities, pathManager } from 'amplify-cli-core';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import { AppSyncCLIInputs } from '../service-walkthrough-types/appsync-user-input-types';
+
+export class AppsyncApiInputState {
```
Comment (edwardfoyle) [inline packages/amplify-category-api/src/provider-utils/awscloudformation/api-input-manager/appsync-api-input-state.ts:6]: maybe too late at this point but would recommend to rename to `AppSyncApiInputState`

File: packages/amplify-category-api/src/index.ts (hunk 4 above)
Comment (jhockett) [reply packages/amplify-category-api/src/index.ts:223]: Prior to this, these errors were being handled in each individual command. This retains mostly the same behavior, while standardizing the precise error handling.

File: packages/amplify-category-api/src/provider-utils/awscloudformation/cdk-stack-builder/apigw-stack-transform.ts [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+            external: true,
+          },
+        });
+
+        try {
+          await sandboxNode.run(overrideCode, overrideJSFilePath).override(this.resourceTemplateObj as AmplifyApigwResourceStack);
+        } catch (err: $TSAny) {
+          const error = new Error(`Skipping override due to ${err}.`);
+          printer.error(`${error}`);
+          error.stack = undefined;
```
Comment (edwardfoyle) [inline packages/amplify-category-api/src/provider-utils/awscloudformation/cdk-stack-builder/apigw-stack-transform.ts:219]: I would keep this stack. Would be useful for customers trying to debug their override files

File: packages/amplify-category-api/src/provider-utils/awscloudformation/cfn-api-artifact-handler.ts [hunk 7]
```diff
@@ -259,10 +267,57 @@ class CfnApiArtifactHandler implements ApiArtifactHandler {
...
+
+    if (!_.isEmpty(serviceConfig.conflictResolution)) {
+      appsyncCLIInputs.serviceConfiguration.conflictResolution = {
+        defaultResolutionStrategy: serviceConfig.conflictResolution.defaultResolutionStrategy,
+        perModelResolutionStrategy: serviceConfig.conflictResolution.perModelResolutionStrategy,
+      };
+    }
+    // deploy appsync inputs
+    const cliState = new AppsyncApiInputState(serviceConfig.apiName);
+    cliState.saveCLIInputPayload(appsyncCLIInputs);
```
Comment (sachscode) [inline packages/amplify-category-api/src/provider-utils/awscloudformation/cfn-api-artifact-handler.ts:297]: Needs Await

File: packages/amplify-category-api/resources/awscloudformation/overrides-resource/AppSync/package.json (hunk 2 above)
Comment (jhockett) [reply packages/amplify-category-api/resources/awscloudformation/overrides-resource/AppSync/package.json:10]: That's a good point, then it should be ^2.3.0, correct?

File: packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthrough-types/aPIGateway-user-input-types.ts [hunk 8]
```diff
@@ -0,0 +1,37 @@
+/**
```
Comment (edwardfoyle) [inline packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthrough-types/aPIGateway-user-input-types.ts:1]: filename is jank

File: packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/apigw-walkthrough.ts [hunk 9]
```diff
@@ -1,190 +1,147 @@
...
-      updatedResult = { answers, dependsOn };
+      updatedResult = { answers };
       break;
     }
     case 'update': {
       const pathToEdit = await inquirer.prompt({
-        name: 'path',
-        message: 'Please select the path you would want to edit',
+        name: 'pathName',
+        message: 'Select the path you would want to edit',
```
Comment (edwardfoyle) [inline packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/apigw-walkthrough.ts:117]:
```suggestion
        message: 'Select the path to edit',
```

File: packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/apigw-walkthrough.ts [hunk 10]
```diff
@@ -1,190 +1,147 @@
...
+  switch (updateApiOperation) {
     case 'add': {
       updatedResult = pathFlow(context, answers);
       break;
     }
     case 'remove': {
       const pathToRemove = await inquirer.prompt({
         name: 'path',
-        message: 'Please select the path you would want to remove',
+        message: 'Select the path you would want to remove',
```
Comment (edwardfoyle) [inline packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/apigw-walkthrough.ts:100]:
```suggestion
        message: 'Select the path to remove',
```

File: packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/apigw-walkthrough.ts [hunk 11]
```diff
@@ -202,111 +164,90 @@ async function askApiNames(context, defaults) {
...
+  currentPath?: ApigwPath,
+): Promise<{ setting?: PermissionSetting; auth?: CrudOperation[]; open?: boolean; userPoolGroups?: $TSObject; guest?: CrudOperation[] }> {
   while (true) {
-    const apiAccess = await inquirer.prompt({
-      name: 'restrict',
-      type: 'confirm',
-      default: !(currentPath && currentPath.open),
-      message: 'Restrict API access',
-    });
+    const apiAccess = await prompter.yesOrNo('Restrict API access', currentPath?.permissions?.setting !== PermissionSetting.OPEN);
```
Comment (edwardfoyle) [inline packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/apigw-walkthrough.ts:181]:
```suggestion
    const apiAccess = await prompter.yesOrNo('Restrict API access?', currentPath?.permissions?.setting !== PermissionSetting.OPEN);
```

File: packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/apigw-walkthrough.ts [hunk 12]
```diff
@@ -202,111 +164,90 @@ async function askApiNames(context, defaults) {
...
-        const permissionSelection = await inquirer.prompt({
-          name: 'selection',
-          type: 'list',
-          message: 'Restrict access by?',
-          choices: ['Auth/Guest Users', 'Individual Groups', 'Both', 'Learn more'],
-          default: 'Auth/Guest Users',
-        });
-
-        permissionSelected = permissionSelection.selection;
+        const permissionSelection = await prompter.pick<'one', string>('Restrict access by?', [
```
Comment (edwardfoyle) [inline packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/apigw-walkthrough.ts:203]:
```suggestion
        const permissionSelection = await prompter.pick<'one', string>('Restrict access by:', [
```

File: packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthrough-types/aPIGateway-user-input-types.ts (hunk 8 above)
Comment (jhockett) [reply packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthrough-types/aPIGateway-user-input-types.ts:1]: intentional

File: packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/apigw-walkthrough.ts (hunk 9 above)
Comment (jhockett) [reply packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/apigw-walkthrough.ts:117]: This is a good suggestion, but I think we should address this in a later PR to avoid breaking tests.

File: packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/apigw-walkthrough.ts [hunk 13]
```diff
@@ -382,69 +315,45 @@ async function ensureAuth(context, apiRequirements, resourceName) {
...
-      }
-    });
-  }
-
-  const crudAnswers = await context.amplify.crudFlow(userType, permissionMap, defaults);
+async function askCRUD(userType: string, permissions: CrudOperation[] = []) {
+  const crudOptions = [CrudOperation.CREATE, CrudOperation.READ, CrudOperation.UPDATE, CrudOperation.DELETE];
+  const crudAnswers = await prompter.pick<'many', string>(`What permissions do you want to grant to ${userType} users?`, crudOptions, {
+    returnSize: 'many',
+    initial: byValues(permissions), // (a, b) => a.toLowerCase() === b.toLowerCase()
```
Comment (edwardfoyle) [inline packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/apigw-walkthrough.ts:341]:
```suggestion
    initial: byValues(permissions),
```

File: packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/apigw-walkthrough.ts (hunk 10 above)
Comment (jhockett) [reply packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/apigw-walkthrough.ts:100]: This is a good suggestion, but I think we should address this in a later PR to avoid breaking tests.

File: packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/apigw-walkthrough.ts (hunk 12 above)
Comment (jhockett) [reply packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/apigw-walkthrough.ts:203]: This change should be fine to merge, the tests do not wait on the `?`

File: packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/apigw-walkthrough.ts [hunk 14]
```diff
@@ -830,12 +664,7 @@ export const openConsole = async (context: $TSContext) => {
...
 
     if (restApis.length > 1) {
-      ({ selectedApi } = await inquirer.prompt({
-        type: 'list',
-        name: 'selectedApi',
-        choices: restApis,
-        message: 'Please select the API',
-      }));
+      selectedApi = await prompter.pick<'one', string>('Select the API', restApis);
     }
```
Comment (edwardfoyle) [inline packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/apigw-walkthrough.ts:668]: nit: prompter.pick will automatically skip the prompt and return the single item if there is only one option

File: packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/apigw-walkthrough.ts [hunk 15]
```diff
@@ -853,34 +682,29 @@ export const openConsole = async (context: $TSContext) => {
...
-            name: 'Elastic Container Service (Deployed container status)',
-            value: elasticContainer,
-          },
-          {
-            name: 'CodePipeline (Container build status)',
-            value: codePipeline,
-          },
-        ],
-      });
+      const selectedConsole = await prompter.pick<'one', string>('Which console you want to open', [
```
Comment (edwardfoyle) [inline packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/apigw-walkthrough.ts:685]:
```suggestion
      const selectedConsole = await prompter.pick<'one', string>('Which console do you want to open?', [
```

File: packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-rds-walkthrough.ts [hunk 16]
```diff
@@ -110,7 +111,7 @@ async function selectCluster(context: $TSContext, inputs, AWS) {
   // Pick first and only value
   const firstCluster = Array.from(clusters.values())[0];
 
-  context.print.info(`${chalk.green('✔')} Only one Cluster was found: '${firstCluster.DBClusterIdentifier}' was automatically selected.`);
+  printer.info(`${chalk.green('✔')} Only one Cluster was found: '${firstCluster.DBClusterIdentifier}' was automatically selected.`);
```
Comment (edwardfoyle) [inline packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-rds-walkthrough.ts:114]: perhaps printer.success() without the bespoke checkmark would be better here?

File: packages/amplify-category-api/tsconfig.json [hunk 17]
```diff
@@ -4,14 +4,17 @@
     "outDir": "lib",
     "rootDir": "src",
     "strict": false, // because package has been converted from js
-    "allowJs": true
+    "allowJs": false
```
Comment (sachscode) [inline packages/amplify-category-api/tsconfig.json:7]: Why is this required ?

File: packages/amplify-category-api/src/provider-utils/awscloudformation/utils/check-appsync-api-migration.ts [hunk 18]
```diff
@@ -0,0 +1,19 @@
+import { $TSContext, AmplifyCategories, getMigrateResourceMessageForOverride } from 'amplify-cli-core';
+import { printer, prompter } from 'amplify-prompts';
+import { AppsyncApiInputState } from '../api-input-manager/appsync-api-input-state';
+import { migrateResourceToSupportOverride } from './migrate-api-override-resource';
+
+export const checkAppsyncApiResourceMigration = async (context: $TSContext, apiName: string, isUpdate: boolean): Promise<boolean> => {
+  const cliState = new AppsyncApiInputState(apiName);
+  if (!cliState.cliInputFileExists()) {
```
Comment (edwardfoyle) [inline packages/amplify-category-api/src/provider-utils/awscloudformation/utils/check-appsync-api-migration.ts:8]: negate this if and early return to prevent nesting

File: packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-inputs-manager/auth-input-state.ts [hunk 19]
```diff
@@ -33,7 +33,7 @@ export class AuthInputState extends CategoryInputState {
   }
 
   public async isCLIInputsValid(cliInputs: CognitoCLIInputs = this.getCLIInputPayload()): Promise<boolean> {
-    const schemaValidator = new CLIInputSchemaValidator('awsCognito', this.#category, 'CognitoCLIInputs');
+    const schemaValidator = new CLIInputSchemaValidator(this.#service, this.#category, 'CognitoCLIInputs');
     return schemaValidator.validateInput(JSON.stringify(cliInputs));
```
Comment (sachscode) [inline packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-inputs-manager/auth-input-state.ts:37]: Do we need to await here , for validation to complete

File: packages/amplify-category-api/src/provider-utils/awscloudformation/utils/print-api-key-warnings.ts [hunk 20]
```diff
@@ -1,16 +1,18 @@
...
+export const printApiKeyWarnings = (oldConfigHadApiKey: boolean, newConfigHasApiKey: boolean) => {
   if (oldConfigHadApiKey && !newConfigHasApiKey) {
-    context.print.warning('The API_KEY auth type has been removed from the API.');
-    context.print.warning(
+    printer.warn('The API_KEY auth type has been removed from the API.');
+    printer.warn(
       'If other resources depend on this API, run "amplify update <category>" and reselect this API to remove the dependency on the API key.',
     );
-    context.print.warning('⚠️  This must be done before running "amplify push" to prevent a push failure');
+    printer.warn('⚠️  This must be done before running "amplify push" to prevent a push failure');
```
Comment (edwardfoyle) [inline packages/amplify-category-api/src/provider-utils/awscloudformation/utils/print-api-key-warnings.ts:10]: this will print a duplicate ⚠️

File: packages/amplify-category-api/src/provider-utils/awscloudformation/utils/print-api-key-warnings.ts [hunk 21]
```diff
@@ -1,16 +1,18 @@
...
   }
 
   if (!oldConfigHadApiKey && newConfigHasApiKey) {
-    context.print.warning('The API_KEY auth type has been added to the API.');
-    context.print.warning(
+    printer.warn('The API_KEY auth type has been added to the API.');
+    printer.warn(
```
Comment (edwardfoyle) [inline packages/amplify-category-api/src/provider-utils/awscloudformation/utils/print-api-key-warnings.ts:15]: same here

File: packages/amplify-category-api/tsconfig.json (hunk 17 above)
Comment (edwardfoyle) [inline packages/amplify-category-api/tsconfig.json:7]: 🎉

File: packages/amplify-category-auth/src/provider-utils/awscloudformation/utils/synthesize-resources.ts [hunk 22]
```diff
@@ -210,32 +233,27 @@ export const createUserPoolGroups = async (context: any, resourceName: string, u
   }
 };
 
-export const updateUserPoolGroups = async (context: any, resourceName: string, userPoolGroupList?: string[]) => {
+export const updateUserPoolGroups = async (context: $TSContext, resourceName: string, userPoolGroupList?: string[]) => {
   if (userPoolGroupList && userPoolGroupList.length > 0) {
     const userPoolGroupPrecedenceList = [];
 
-    for (let i = 0; i < userPoolGroupList.length; i += 1) {
+    for (let i = 0; i < userPoolGroupList.length; ++i) {
```
Comment (edwardfoyle) [inline packages/amplify-category-auth/src/provider-utils/awscloudformation/utils/synthesize-resources.ts:240]: would be nice to convert to for each loop

**Review by sachscode — COMMENTED** [review]
LGTM, some async function questions

File: packages/amplify-cli-core/tsconfig.json [hunk 23]
```diff
@@ -3,5 +3,6 @@
   "compilerOptions": {
     "outDir": "./lib",
     "rootDir": "./src",
+    "useUnknownInCatchVariables":false
```
Comment (edwardfoyle) [inline packages/amplify-cli-core/tsconfig.json:6]: won't this get pulled in from the base tsconfig?

File: packages/amplify-container-hosting/lib/ElasticContainer/index.js [hunk 24]
```diff
@@ -228,7 +228,7 @@ export async function generateHostingResources(
 
   const projectBackendDirPath = context.amplify.pathManager.getBackendDirPath();
 
-  /** @type {import('amplify-category-api').ApiResource & {service: string, domain: string, providerPlugin:string, hostedZoneId: string, iamAccessUnavailable: boolean}} */
+  /** @type {import('@aws-amplify/amplify-category-api').ApiResource & {service: string, domain: string, providerPlugin:string, hostedZoneId: string, iamAccessUnavailable: boolean}} */
```
Comment (edwardfoyle) [inline packages/amplify-container-hosting/lib/ElasticContainer/index.js:231]: remove?

File: packages/amplify-e2e-core/src/init/overrideStack.ts [hunk 25]
```diff
@@ -35,3 +35,20 @@ export function amplifyOverrideAuth(cwd: string, settings: {}) {
...
 }
+
+export function amplifyOverrideApi(cwd: string, settings: any) {
+  return new Promise((resolve, reject) => {
+    const args = ['override', 'api'];
+    const chain = spawn(getCLIPath(), args, { cwd, stripColors: true });
+    if (settings.isMigratedProject === true) {
+      chain.wait('Do you want to edit override.ts file now?').sendConfirmNo().sendEof();
+    }
+    chain.run((err: Error) => {
```
Comment (edwardfoyle) [inline packages/amplify-e2e-core/src/init/overrideStack.ts:46]: nit: would be nice to use chain.runAsync() instead

File: packages/amplify-graphiql-explorer/tsconfig.json [hunk 26]
```diff
@@ -19,7 +19,8 @@
     "noEmit": true,
     "jsx": "react-jsx",
     "noImplicitAny": false,
-    "noFallthroughCasesInSwitch": true
+    "noFallthroughCasesInSwitch": true,
+    "useUnknownInCatchVariables":false,
```
Comment (edwardfoyle) [inline packages/amplify-graphiql-explorer/tsconfig.json:23]: shouldn't this be pulled in from the base config?

File: packages/amplify-graphql-function-transformer/src/__tests__/amplify-graphql-function-transformer-override.test.ts [hunk 27]
```diff
@@ -0,0 +1,29 @@
+'use strict';
```
Comment (edwardfoyle) [inline packages/amplify-graphql-function-transformer/src/__tests__/amplify-graphql-function-transformer-override.test.ts:1]: this shouldn't be needed

File: packages/amplify-graphql-http-transformer/src/__tests__/amplify-graphql-http-transformer-override.test.ts [hunk 28]
```diff
@@ -0,0 +1,31 @@
+'use strict';
```
Comment (edwardfoyle) [inline packages/amplify-graphql-http-transformer/src/__tests__/amplify-graphql-http-transformer-override.test.ts:1]: same here and anywhere else we have this in a .ts file

File: packages/amplify-provider-awscloudformation/src/download-api-models.ts [hunk 29]
```diff
@@ -18,21 +19,21 @@ export async function downloadAPIModels(context: $TSContext, allResources) {
   const promises = [];
 
   if (resources.length > 0) {
-    context.print.info('\nCreating API models...');
+    printer.info('\nCreating API models...');
```
Comment (edwardfoyle) [inline packages/amplify-provider-awscloudformation/src/download-api-models.ts:22]:
```suggestion
    printer.blankLink();
    printer.info('Creating API models...');
```

File: packages/amplify-category-api/src/provider-utils/awscloudformation/apigw-input-state.ts [hunk 30]
```diff
@@ -0,0 +1,247 @@
...
+    this.resourceName = this.resourceName ?? adminQueriesProps.apiName;
+    if (!(await prompter.yesOrNo(getMigrateResourceMessageForOverride(AmplifyCategories.API, this.resourceName, true), true))) {
+      return;
+    }
+    const resourceDirPath = pathManager.getResourceDirectoryPath(this.projectRootPath, AmplifyCategories.API, this.resourceName);
+
+    this.context.filesystem.remove(join(resourceDirPath, PathConstants.ParametersJsonFileName));
+    this.context.filesystem.remove(join(resourceDirPath, 'admin-queries-cloudformation-template.json'));
+
+    return this.updateAdminQueriesResource(adminQueriesProps);
```
[the remark this answers is not in the document]
Comment (jhockett) [reply packages/amplify-category-api/src/provider-utils/awscloudformation/apigw-input-state.ts:114]: No, it's an async function so we're still returning a promise. The caller will do the awaiting

File: packages/amplify-category-api/src/commands/api/push.ts [hunk 31]
```diff
@@ -0,0 +1,11 @@
+import { $TSContext, AmplifyCategories } from 'amplify-cli-core';
+
+const subcommand = 'push';
+
+export const name = subcommand;
+
+export const run = async (context: $TSContext) => {
+  const resourceName = context.parameters.first;
+  context.amplify.constructExeInfo(context);
+  return context.amplify.pushResources(context, AmplifyCategories.API, resourceName);
```
[the remark this answers is not in the document]
Comment (jhockett) [reply packages/amplify-category-api/src/commands/api/push.ts:10]: No, it's an async function so we're still returning a promise. The caller will do the awaiting

File: packages/amplify-category-api/resources/awscloudformation/overrides-resource/AppSync/package.json (hunk 2 above)
Comment (edwardfoyle) [reply packages/amplify-category-api/resources/awscloudformation/overrides-resource/AppSync/package.json:10]: it should be whatever is in the package.json of cli-extensibility-helper. learna will do the rest

File: packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-inputs-manager/auth-input-state.ts (hunk 19 above)
Comment (jhockett) [reply packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-inputs-manager/auth-input-state.ts:37]: Same as above, it's fine because a Promise is returned

File: packages/amplify-category-api/src/commands/api/push.ts (hunk 31 above)
[the remark this answers is not in the document]
Comment (edwardfoyle) [reply packages/amplify-category-api/src/commands/api/push.ts:10]: it's returning the promise and the run is awaited, so I think it should be good

File: packages/amplify-container-hosting/lib/ElasticContainer/index.js (hunk 24 above)
Comment (jhockett) [reply packages/amplify-container-hosting/lib/ElasticContainer/index.js:231]: Why? The package name is now `@aws-amplify/amplify-category-api`

File: packages/amplify-category-api/src/provider-utils/awscloudformation/cfn-api-artifact-handler.ts (hunk 7 above)
Comment (jhockett) [reply packages/amplify-category-api/src/provider-utils/awscloudformation/cfn-api-artifact-handler.ts:297]: @akshbhu this looks like a bug to me

File: packages/amplify-category-api/resources/awscloudformation/overrides-resource/AppSync/package.json (hunk 2 above)
Comment (jhockett) [reply packages/amplify-category-api/resources/awscloudformation/overrides-resource/AppSync/package.json]: @edwardfoyle Will lerna make changes if these pacakge.json's are in the `...category-api/resources/` directory?

File: packages/amplify-cli-core/tsconfig.json (hunk 23 above)
Comment (akshbhu) [reply packages/amplify-cli-core/tsconfig.json:6]: somehow it isnt , I tried before

File: packages/amplify-e2e-core/src/init/overrideStack.ts (hunk 25 above)
Comment (akshbhu) [reply packages/amplify-e2e-core/src/init/overrideStack.ts:46]: ill keep it same for now as all tests are using this

File: packages/amplify-graphiql-explorer/tsconfig.json (hunk 26 above)
Comment (akshbhu) [reply packages/amplify-graphiql-explorer/tsconfig.json:23]: same as above

**Review by sachscode — APPROVED** [review]

File: packages/amplify-container-hosting/lib/ElasticContainer/index.js (hunk 24 above)
Comment (edwardfoyle) [reply packages/amplify-container-hosting/lib/ElasticContainer/index.js:231]: Oh I see, this is some sort of comment to resolve types in js files. nvm
